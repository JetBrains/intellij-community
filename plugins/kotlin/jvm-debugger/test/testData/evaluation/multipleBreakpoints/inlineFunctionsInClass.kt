package inlineFunctionsInClass

class A {
    inline fun foo(fooParam: Int) {
        val xFoo = 1
        bar(2)
        //Breakpoint!
        println()
    }

    inline fun bar(barParam: Int) {
        val xBar = 2
        baz(3)
        //Breakpoint!
        println()
    }

    inline fun baz(bazParam: Int) {
        val xBaz = 3
        //Breakpoint!
        println()
    }
}

fun main() {
    val a = A()
    a.foo(1)
    a.foo(2)
}

// EXPRESSION: xBaz + bazParam
// RESULT: 6: I
// EXPRESSION: xBar + barParam
// RESULT: 4: I
// EXPRESSION: xFoo + fooParam
// RESULT: 2: I

// EXPRESSION: xBaz + bazParam
// RESULT: 6: I
// EXPRESSION: xBar + barParam
// RESULT: 4: I
// EXPRESSION: xFoo + fooParam
// RESULT: 3: I

// PRINT_FRAME
// SHOW_KOTLIN_VARIABLES
