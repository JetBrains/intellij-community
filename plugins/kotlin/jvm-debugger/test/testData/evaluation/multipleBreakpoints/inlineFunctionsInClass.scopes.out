LineBreakpoint created at inlineFunctionsInClass.kt:8
LineBreakpoint created at inlineFunctionsInClass.kt:15
LineBreakpoint created at inlineFunctionsInClass.kt:21
Run Java
Connected to the target VM
inlineFunctionsInClass.kt:21
Compile bytecode for xBaz + bazParam
InlineStackFrame baz (inlineFunctionsInClass.kt:21)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] bazParam: int = 3 (inlineFunctionsInClass.kt:18)
    JavaValue[local] xBaz: int = 3 (inlineFunctionsInClass.kt:19)
InlineStackFrame bar (inlineFunctionsInClass.kt:13)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] barParam: int = 2
    JavaValue[local] xBar: int = 2
InlineStackFrame foo (inlineFunctionsInClass.kt:6)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 1
    JavaValue[local] xFoo: int = 1
KotlinStackFrame (inlineFunctionsInClass.kt:27)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
inlineFunctionsInClass.kt:15
Compile bytecode for xBar + barParam
InlineStackFrame bar (inlineFunctionsInClass.kt:15)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] barParam: int = 2 (inlineFunctionsInClass.kt:11)
    JavaValue[local] xBar: int = 2 (inlineFunctionsInClass.kt:12)
InlineStackFrame foo (inlineFunctionsInClass.kt:6)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 1
    JavaValue[local] xFoo: int = 1
KotlinStackFrame (inlineFunctionsInClass.kt:27)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
inlineFunctionsInClass.kt:8
Compile bytecode for xFoo + fooParam
InlineStackFrame foo (inlineFunctionsInClass.kt:8)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 1 (inlineFunctionsInClass.kt:4)
    JavaValue[local] xFoo: int = 1 (inlineFunctionsInClass.kt:5)
KotlinStackFrame (inlineFunctionsInClass.kt:27)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
inlineFunctionsInClass.kt:21
InlineStackFrame baz (inlineFunctionsInClass.kt:21)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] bazParam: int = 3 (inlineFunctionsInClass.kt:18)
    JavaValue[local] xBaz: int = 3 (inlineFunctionsInClass.kt:19)
InlineStackFrame bar (inlineFunctionsInClass.kt:13)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] barParam: int = 2
    JavaValue[local] xBar: int = 2
InlineStackFrame foo (inlineFunctionsInClass.kt:6)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 2
    JavaValue[local] xFoo: int = 1
KotlinStackFrame (inlineFunctionsInClass.kt:28)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
inlineFunctionsInClass.kt:15
InlineStackFrame bar (inlineFunctionsInClass.kt:15)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] barParam: int = 2 (inlineFunctionsInClass.kt:11)
    JavaValue[local] xBar: int = 2 (inlineFunctionsInClass.kt:12)
InlineStackFrame foo (inlineFunctionsInClass.kt:6)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 2
    JavaValue[local] xFoo: int = 1
KotlinStackFrame (inlineFunctionsInClass.kt:28)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
inlineFunctionsInClass.kt:8
InlineStackFrame foo (inlineFunctionsInClass.kt:8)
    JavaValue[local] this: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
    JavaValue[local] fooParam: int = 2 (inlineFunctionsInClass.kt:4)
    JavaValue[local] xFoo: int = 1 (inlineFunctionsInClass.kt:5)
KotlinStackFrame (inlineFunctionsInClass.kt:28)
    JavaValue[local] a: inlineFunctionsInClass.A = inlineFunctionsInClass.A@hashCode
        DummyMessageValueNode = Class has no properties
Disconnected from the target VM

Process finished with exit code 0
